pkg/astvisitor/visitor.go

```diff
@@ -1349,6 +1350,10 @@ func (w *Walker) appendAncestor(ref int, kind ast.NodeKind) {
 		default:
 			return
 		}
+		if len(typeName) == 0 {
```

Comment:
Can this even happen? If `operationType` does not match with any case it returns. Could you add a test?

Reply:
Maybe we should return an error in the `default:` case. I'll take a look around. I'll add a test too.

Reply:
I've looked around and it seems we can safely change that to stop parsing with an error if the operation is not valid in that switch. I've also added a test for the new error that this PR introduces. PTAL

Reply:
Much better. Thanks for addressing it.

Reply:
🫡